packages/vue-inertia/src/index.ts

```diff
@@ -1,11 +1,42 @@
-import { Config, NamedInputEvent, RequestMethod, SimpleValidationErrors, toSimpleValidationErrors, ValidationConfig, ValidationErrors, resolveUrl, resolveMethod } from 'laravel-precognition'
+import {
+    Config,
+    NamedInputEvent,
+    RequestMethod,
+    SimpleValidationErrors,
+    toSimpleValidationErrors,
+    ValidationConfig,
+    ValidationErrors,
+    resolveUrl,
+    resolveMethod,
+    Validator,
+} from 'laravel-precognition'
 import { useForm as usePrecognitiveForm, client } from 'laravel-precognition-vue'
-import { useForm as useInertiaForm } from '@inertiajs/vue3'
+import { InertiaForm, useForm as useInertiaForm } from '@inertiajs/vue3'
 import { watchEffect } from 'vue'
 
 export { client }
 
-export const useForm = <Data extends Record<string, unknown>>(method: RequestMethod|(() => RequestMethod), url: string|(() => string), inputs: Data, config: ValidationConfig = {}): any => {
+type Precognitive<Data extends object> = {
+    validating: boolean
+    touched: (name: keyof Data) => boolean
+    valid: (name: keyof Data) => boolean
+    invalid: (name: keyof Data) => boolean
+    clearErrors(...names: string[]): Form<Data>
+    reset(...names: string[]): void
+    setErrors(errors: SimpleValidationErrors | ValidationErrors): Form<Data>
+    forgetError(name: string | NamedInputEvent): Form<Data>
+    setError(key: any, value?: any): Form<Data>
+    transform(callback: (data: Data) => Record<string, unknown>): Form<Data>
+    validate(name?: string | NamedInputEvent): Form<Data>
+    setValidationTimeout(duration: number): Form<Data>
+    validateFiles(): Form<Data>
+    submit(submitMethod: RequestMethod | Config, submitUrl?: string, submitOptions?: any): void
+    validator(): Validator
+};
+
+export type Form<Data extends object> = InertiaForm<Data> & Precognitive<Data>
```

Comment:
I split the definition so that below, we could ensure the second object in the `Object.assign` call was valid.